packages/macro/test/js-useLingui.ts

```diff
@@ -0,0 +1,380 @@
+
+  console.log(i18n);
+  const a = t\`Text\`;
+}
+    `,
+    expected: `
+import { useLingui } from "@lingui/react";
+function MyComponent() {
+  const { i18n, _: _t } = useLingui();
+  console.log(i18n);
+  const a = _t(
+    /*i18n*/
+    {
+      id: "xeiujy",
+      message: "Text",
+    }
+  );
+}
+`,
+  },
+  {
+    name: "work with existing useLingui statement",
+    input: `
+import { useLingui as useLinguiMacro } from '@lingui/macro';
+import { useLingui } from '@lingui/react';
+
+function MyComponent() {
+  const { _ } = useLingui();
+
+  console.log(_);
+  const { t } = useLinguiMacro();
+  const a = t\`Text\`;
+}
+    `,
+    expected: `
+import { useLingui } from "@lingui/react";
+function MyComponent() {
+  const { _ } = useLingui();
+  console.log(_);
+  const { _: _t } = useLingui();
+  const a = _t(
+    /*i18n*/
+    {
+      id: "xeiujy",
+      message: "Text",
+    }
+  );
+}
+`,
+  },
+
+  {
+    // todo: implement this
+    skip: true,
+    name: "work with renamed existing useLingui statement",
+    input: `
+import { useLingui as useLinguiMacro } from '@lingui/macro';
+import { useLingui as useLinguiRenamed  } from '@lingui/react';
+
+function MyComponent() {
+  const { _ } = useLinguiRenamed();
+
+  console.log(_);
+  const { t } = useLinguiMacro();
+  const a = t\`Text\`;
+}
+    `,
+    expected: `
+import { useLingui as useLinguiRenamed  } from '@lingui/react';
+import { useLingui } from "@lingui/react";
+function MyComponent() {
+  const { _ } = useLinguiRenamed();
+  console.log(_);
+  const { _: _t } = useLingui();
+  const a = _t(
+    /*i18n*/
+    {
+      id: "xeiujy",
+      message: "Text",
+    }
+  );
+}
+`,
+  },
+  {
+    name: "work with multiple react components",
+    input: `
+import { useLingui } from '@lingui/macro';
+
+function MyComponent() {
+  const { t } = useLingui();
+  const a = t\`Text\`;
+}
+
+function MyComponent2() {
+  const { t } = useLingui();
+  const b = t\`Text\`;
+}`,
+    expected: `
+import { useLingui } from "@lingui/react";
+function MyComponent() {
+  const { _: _t } = useLingui();
+  const a = _t(
+    /*i18n*/
+    {
+      id: "xeiujy",
+      message: "Text",
+    }
+  );
+}
+function MyComponent2() {
+  const { _: _t } = useLingui();
+  const b = _t(
+    /*i18n*/
+    {
+      id: "xeiujy",
+      message: "Text",
+    }
+  );
+}
+`,
+  },
+  {
+    name: "support configuring runtime module import using LinguiConfig.runtimeConfigModule",
+    macroOpts: {
+      linguiConfig: makeConfig(
+        {
+          runtimeConfigModule: {
+            useLingui: ["@my/lingui-react", "myUselingui"],
+          },
+        },
+        { skipValidation: true }
+      ),
+    },
+    input: `
+import { useLingui } from '@lingui/macro';
+
+function MyComponent() {
+  const { t } = useLingui();
+  const a = t\`Text\`;
+}
+`,
+    expected: `
+import { myUselingui } from "@my/lingui-react";
+function MyComponent() {
+  const { _: _t } = useLingui();
```

Comment:
just found that this is incorrect, it imports `myUselingui` but uses `useLingui`